components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2721,7 +2721,7 @@ private boolean isMemberAttributeRequired(Map<String, Boolean> requiredAttribute
     }
 
     @Override
-    public void deleteGroup(String groupId) throws NotFoundException, CharonException {
+    public void deleteGroup(String groupId) throws NotFoundException, CharonException, BadRequestException {
```

Comment:
Changing the public method will lead to some issues with the exsiting usages. The best way I think we can create a separate method with the same impl and add the extra exception.
@senthalan any input on this.

Reply:
yes. we can't change the signature of a public method. Can't we throw this as a CharonException with a specific error code and then process in the API layer. 

How we are handling the scim disabled userstores case